### PR TITLE
New interceptors and updates to existing ones

### PR DESCRIPTION
**Pull request by agigao**
### CORS headers interceptor

1. The allowed origin is configurable from env variables and in state it looks for `:cors-url` under `:deps` key
2. It also supports `preflight` requests

Error response
Error interceptor wrapped in `bad request` response instead of 200

### Multipart/form-data support

1. Tiny but subtle improvement of params interceptor
2. Mock and Peridot library used for testing purposes as it allows us to simply mock multipart requests, and many others, but that's what we needed at the moment.

### Coercion improvements

1. The first change I propose here is avoiding nesting coercion result under param type (path, query, form) and just keeping it under params
2. Validation error returns humanized details

### kebab-camel request/request - experimental feature, up for discussion

If the back-end is Clojure and Front-end Javascript we encounter inconsistency of keys in the context of the language.
Hence the experimental solution that we can discuss:

1. kebab-camel interceptor transforms each key upon request to kebab:
- original request `{:request {:params {:paramKey1 1 :paramKey2 2 :paramKey3 3}}}`
- request after interceptor `{:request {:params {:param-key-1 1 :param-key-2 2 :param-key-3 3}}}`
          
2. The keys in the response body are also transformed from kebab to camel:
- response before interceptor - `{:response {:body {:param-key-1 1 :param-key-2 2 :param-key-3 3}}}`
- after `{:response {:body {:paramKey1 1 :paramKey2 2 :paramKey3 3}}}`

P.S. `slf4j-simple` is also added to `deps.edn` to finally get rid of warning at the beginning of http server.

### Tester info
All tests are provided

### Completion Checklist

- [ ] Add description of the changes made here to the changelog file
> Will be added after approval
- [ ] Update the documentation as necessary
> The same here

### CODE REVIEW

**Comment by agigao** [comment]
PR included wrapping `error` in `ring/bad-request` but it caused havoc in the tests.

**Review by gmsvalente — APPROVED** [review]
LGTM!

**Review by soulflyer — COMMENTED** [review]
Does this really need discussion? Seems like a useful thing to have around, and adding it won't alter anything unless the user chooses to use it.
If it really does need discussion then it should be a separate PR

**Comment by soulflyer** [comment]
Should probably be 4 PRs, (This is only a comment, not a request for change)

File: src/xiana/interceptor.clj [hunk 1]
```diff
@@ -143,42 +145,36 @@
...
-                         (:path schemas)
-                         (assoc :path (valid? (:path schemas) path))
-
-                         (:query schemas)
-                         (assoc :query (valid? (:query schemas) query))
-
-                         (:form schemas)
-                         (assoc :form (valid? (:form schemas) form-params)))]
+                    schemas (or (get-in state [:request-data :match :data method :parameters])
+                                (get-in state [:request-data :match :data :parameters]))
```
Comment (g-krisztian) [inline src/xiana/interceptor.clj:174]: I'd keep the merge as it was, to be able to define the path related parameters once, and method parameters separately.

**Comment by agigao** [comment]
> Should probably be 4 PRs, (This is only a comment, not a request for change)

agreed.

File: src/xiana/interceptor.clj (hunk 1 above)
Comment (agigao) [reply src/xiana/interceptor.clj]: Ok.

File: src/xiana/interceptor.clj [hunk 2]
```diff
@@ -143,42 +145,36 @@
...
+                    cc (cond
+                         (:path schemas) (valid? (:path schemas) path)
+                         (:query schemas) (valid? (:query schemas) query)
+                         (:form schemas) (valid? (:form schemas) form-params))]
```
[the remark this answers is not in the document]
Comment (agigao) [reply src/xiana/interceptor.clj]: Yup, the change undone.

**Pull request merged**